agent/agent.go

```diff
@@ -115,7 +145,15 @@ func handlePullRequest(token string) func(interface{}, webhooks.Header) {
 			if len(contents) > 0 {
 				err = kubectl.Apply(dir, strings.Join(contents, "---\n"))
 				if err != nil {
-					log.Fatalf("error applying manifests to the cluster: %v\n", err)
+					err1 := updateStatus("error", fmt.Sprintf("deployment of %s failed: %v", dir, err))
+					if err1 != nil {
+						log.Fatalf("error updating status %v\n", err1)
+					}
+				} else {
+					err1 := updateStatus("success", fmt.Sprintf("deployment of %s succeeded", dir))
+					if err1 != nil {
+						log.Fatalf("error updating status %v\n", err1)
+					}
```

Comment:
Is there any way of reducing the nesting around this part...?

Reply:
Thanks for the review! Yeah, going to refactor that when we add the dry-run and master merge